<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>840</width>
    <height>750</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QString常用功能</string>
  </property>
  <widget class="QFrame" name="frame_buttons">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>20</y>
     <width>351</width>
     <height>711</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::Shape::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Shadow::Raised</enum>
   </property>
   <widget class="QGroupBox" name="groupBox_storage_related">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>170</y>
      <width>331</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>存储相关</string>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0">
      <widget class="QPushButton" name="pushButton_isNull_and_isEmpty">
       <property name="text">
        <string>isNull &amp;&amp; isEmpty</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QPushButton" name="pushButton_resize">
       <property name="text">
        <string>resize</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="pushButton_size_and_length">
       <property name="text">
        <string>size &amp;&amp; length</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QPushButton" name="pushButton_fill">
       <property name="text">
        <string>fill</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox_string_interception">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>331</width>
      <height>151</height>
     </rect>
    </property>
    <property name="title">
     <string>字符串截取</string>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QPushButton" name="pushButton_front_and_back">
       <property name="text">
        <string>front &amp;&amp; back</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QPushButton" name="pushButton_sliced">
       <property name="text">
        <string>sliced *</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="pushButton_first_and_last">
       <property name="text">
        <string>first &amp;&amp; last</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QPushButton" name="pushButton_section">
       <property name="text">
        <string>section *</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox_search_and_judgement">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>350</y>
      <width>331</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>搜索和判断</string>
    </property>
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="0" column="1">
      <widget class="QPushButton" name="pushButton_contains">
       <property name="text">
        <string>contains</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QPushButton" name="pushButton_indexOf_and_lastIndexOf">
       <property name="text">
        <string>indexOf &amp;&amp; lastIndexOf</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="pushButton_endsWith_and_startsWith">
       <property name="text">
        <string>endsWith &amp;&amp; startsWith</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QPushButton" name="pushButton_count">
       <property name="text">
        <string>count</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox_conversion_or_modification">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>530</y>
      <width>331</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>转换或修改</string>
    </property>
    <layout class="QGridLayout" name="gridLayout_4">
     <item row="0" column="1">
      <widget class="QPushButton" name="pushButton_insert">
       <property name="text">
        <string>insert</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QPushButton" name="pushButton_trimmed_and_simplified">
       <property name="text">
        <string>trimmed &amp;&amp; simplified</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="pushButton_remove">
       <property name="text">
        <string>remove</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QPushButton" name="pushButton_replace">
       <property name="text">
        <string>replace</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QFrame" name="frame_input_and_display">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>20</y>
     <width>441</width>
     <height>711</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::Shape::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Shadow::Raised</enum>
   </property>
   <widget class="QGroupBox" name="groupBox_input">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>421</width>
      <height>171</height>
     </rect>
    </property>
    <property name="title">
     <string>输入</string>
    </property>
    <widget class="QLabel" name="label_str1">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>40</y>
       <width>41</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>str1</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_str2">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>90</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>/</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_str2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>41</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>str2</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_str1">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>40</y>
       <width>331</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>~/source/Qt/Application/cpp/learning</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>~/test.cpp</string>
      </property>
     </item>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_result_display">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>180</y>
      <width>421</width>
      <height>511</height>
     </rect>
    </property>
    <property name="title">
     <string>结果显示</string>
    </property>
    <layout class="QGridLayout" name="gridLayout_5">
     <item row="0" column="0">
      <widget class="QPushButton" name="pushButton_clear">
       <property name="text">
        <string>清空文本框</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPlainTextEdit" name="plainTextEdit"/>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
